ipyparallel/client/asyncresult.py

```diff
@@ -678,15 +678,11 @@ def _yield_child_results(self, child):
         for use in iterator methods
         """
         rlist = child.result()
+        if not isinstance(rlist, list):
```

Comment:
Is there any case it wouldn't be a `list`? Just thinking whether `collections.Sequence` might be more appropriate or not.